File: 220/ruby/template_method/encoder_spec.rb
require './encoder'

describe ReverseEncoder do
  it "reverses each word in the input" do
    encoder = ReverseEncoder.new
    encoder.encode("abcde 123").should == "edcba 321" 
  end
end

describe DoublingEncoder do
  it "repeats each word in the input" do
    encoder = DoublingEncoder.new
    encoder.encode("abcde 123").should == "abcdeabcde 123123" 
  end
end

File: 220/js/template_method/encoder_spec.js
require('./encoder')

describe("Reverse Encoding", function() {
  it("reverses each word in the input", function() {
    expect(new ReverseEncoder().encode("abcde 123")).toEqual("edcba 321");
  });
});

describe("Double Encoding", function() {
  it("repeats each word in the input", function() {
    expect(new DoubleEncoder().encode("abcde 123")).toEqual("abcdeabcde 123123");
  });
});
